### PR TITLE
add --log-shim option to allow external containerd-logging-plugin

### PR DESCRIPTION
**Pull request by phuslu**
for https://github.com/containerd/nerdctl/issues/2687

### CODE REVIEW

File: cmd/nerdctl/container_create.go [hunk 1]
```diff
@@ -356,6 +356,10 @@ func processContainerCreateOptions(cmd *cobra.Command) (opt types.ContainerCreat
 	if err != nil {
 		return
 	}
+	opt.LogShim, err = cmd.Flags().GetString("log-shim")
```
Comment (Zheaoli) [inline cmd/nerdctl/container_create.go:359]: You need add the flag here https://github.com/containerd/nerdctl/blob/main/cmd/nerdctl/container_run.go#L75 first

Comment (AkihiroSuda) [reply cmd/nerdctl/container_create.go:359]: - I feel the CLI should be like `--log-driver uri --log-opt uri=<uri>`, or `--log-driver binary --log-opt binary=/usr/local/bin/...`
- Please update https://github.com/containerd/nerdctl/blob/main/docs/command-reference.md
- Please add tests
- Please consider making this experimental https://github.com/containerd/nerdctl/blob/8af72334f0e73b01a9c85a746c15ac0e5772aad7/pkg/cmd/image/convert.go#L214 https://github.com/containerd/nerdctl/blob/main/docs/experimental.md

File: docs/command-reference.md [hunk 2]
```diff
@@ -347,6 +347,8 @@ Logging flags:
       - :whale: `--log-opt=tag=<VALUE>`: A string that is appended to the
           `APP-NAME` in the `syslog` message. By default, nerdctl uses the first
           12 characters of the container ID to tag log messages.
+  - :whale: `--log-driver=binary`: Pipes log messages to `binary`. The `binary` daemon must be running on the host machine.
+    - :whale: `--log-opt=binary=<BINARY_PATH>`: Specify binary path.
   - :nerd_face: Accepts a LogURI which is a containerd shim logger. A scheme must be specified for the URI. Example: `nerdctl run -d --log-driver binary:///usr/bin/ctr-journald-shim docker.io/library/hello-world:latest`. An implementation of shim logger can be found at (<https://github.com/containerd/containerd/tree/dbef1d56d7ebc05bc4553d72c419ed5ce025b05d/runtime/v2#logging>)
```
Comment (AkihiroSuda) [inline docs/command-reference.md:352]: Looks like we already had this feature

**Comment by phuslu** [comment]
Seems it is, let me close this PR first and learn about it.

**Pull request closed without merging**